src/Microsoft.TestPlatform.CommunicationUtilities/Serialization/TestCaseConverter.cs

```diff
@@ -71,7 +71,9 @@ public override object ReadJson(JsonReader reader, Type objectType, object exist
                             testCase.LineNumber = int.Parse(propertyData); break;
                         default:
                             // No need to register member properties as they get registered as part of TestCaseProperties class.
-                            TestProperty.Register(testProperty.Id, testProperty.Label, testProperty.GetValueType(), typeof(TestObject));
+                            // Use a previously registered property if one exists.
+                            testProperty = TestProperty.Find(testProperty.Id) ??
```

Comment:
Register already does a find internally, & then adds the property if not present, do we need do an explicit find?

Reply:
I had taken a look and it seemed like it would add the owner to a hashset if the supplied owner is different. I was not sure that is necessary since the owner doesn’t seem to be used by anything...

It also seemed cleaner and more explicit to do a Find to avoid confusion (such as what if the property is already registered with different attributes in the process?)

I can certainly omit the call to Find if you disagree.